src/sys/socket/addr.rs

```diff
@@ -973,4 +1064,294 @@ pub mod sys_control {
             write!(f, "id: {} unit: {}", self.id(), self.unit())
         }
     }
+
+    impl fmt::Debug for SysControlAddr {
+        fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+            fmt::Display::fmt(self, f)
+        }
+    }
+}
+
+
+#[derive(Clone, Copy, Eq, Hash, PartialEq)]
+pub struct EtherAddr {
```

Comment:
Why didn't you make this a newtype around the underlying `libc` type as I asked? You add a lot of types that I don't think are necessary.

```rust
cfg_if! {
    if #[cfg(any(target_os = "android", target_os = "linux"))] {
        pub struct EtherAddr(libc::sockaddr_ll);
    else if #[cfg(any(target_os = "dragonfly",
          target_os = "freebsd",
          target_os = "ios",
          target_os = "macos",
          target_os = "netbsd",
          target_os = "openbsd"))] {
        pub struct EtherAddr(libc::sockaddr_dl);
}
```

Then to access the MAC address, you can implement a `EtherAddr.mac()` or something like that. You can defer on all the other struct contents and just implement the MAC address part for now.

Reply:
Code:

```rust
#[cfg(any(target_os = "android", target_os = "linux"))]
pub struct EtherAddr(libc::sockaddr_ll);

#[cfg(any(target_os = "dragonfly",
          target_os = "freebsd",
          target_os = "ios",
          target_os = "macos",
          target_os = "netbsd",
          target_os = "openbsd"))]
pub struct EtherAddr(libc::sockaddr_dl);

impl EtherAddr {
    pub fn addr(&self) -> [u8; 6];
}
```

is that ok ?

Reply:
Yes, that's more what I was thinking. Is there any other data in there that's common across platforms that we could expose besides addr()?

Reply:
just hwaddr (`[u8; 6]`) . no more other data.

i hope u know i havn't english skill.

sry for that.

Reply:
@LuoZijun No worries about your language skills, we seem to be communicating just fine! And then that's fine if that's the only data there, I think we're on the right track for a nice API here!